module/modmanager/manager.go

```diff
@@ -963,9 +1027,11 @@ func (m *module) cleanupAfterStartupFailure(mgr *Manager, afterCrash bool) {
 	if afterCrash {
 		for r, mod := range mgr.rMap {
 			if mod == m {
+				// TODO: lock or partition!
```

Comment:
still have TODOs?

Reply:
dang, yeah I forgot about this one - looks likes our existing tests don't catch races related to this.

Reply:
addressed by breaking the cleanup function into two: https://github.com/viamrobotics/rdk/pull/3641/commits/e8217d9886aa77b7f86fb3c5d4d9486c16b89017 - the non-crash version called here should be thread-safe, and the "crash" version is only called in contexts where we are holding a write lock.